### DISCUSSION TITLE
Communicating between two boards with a single data wire. (half duplex)

### DISCUSSION THREAD

**Discussion by LazaroFilm** [earlephilhower/arduino-pico#1739] [General]
I have a situation where I need two boards to communicate but there is only one data wire available between the two. Both systems are powered from the same battery so they have common ground. Is this possible with arduino-pico?

**Comment by earlephilhower** [comment]
For bidirectional communications, Look up AlohaNet CSMA.  You'll need to implement it yourself unless there a library already written for arduino.

Unidirectional comms, just use a UART.

**Reply by maxgerhardt** [reply]
And even UART can be bidirectional when TX and RX are connected together with a small circuit on each side, see e.g. [this](https://ww1.microchip.com/downloads/en/AppNotes/USART-in-One-Wire-Mode-ApplicationNote-DS00002658.pdf) chapter 3.1. You just have to make sure it's clear at all times who can send so that noone drives the bus at the same time.

Also, OneWire master + slave libraries should be [widely available](https://forum.arduino.cc/t/arduino-owi-1-wire-master-slave-library/486185).
